src/gobapi/storage.py

```diff
@@ -171,11 +171,40 @@ def _get_convert_for_table(table, filter={}):
     """
     def convert(entity):
         # Use the sqltypes to get the correct gobtype and return a dict
-        return {column.name: _to_gob_value(entity, column.name, type(column.type)) for column in columns}
+        hal_entity = {column.name: _to_gob_value(entity, column.name, type(column.type)) for column in columns}
 
-    # Get all metadata fields and filter them from the columns returned by the database view
+        # Add references to other entities
+        if references:
+            hal_entity['_embedded'] = {
+                v['attribute_name']: _create_reference(entity, k, v) for k, v in references.items()
+            }
+        return hal_entity
+
+    # Get all metadata or reference fields and filter them from the columns returned by the database view
     metadata_column_list = [k for k in filter.keys()]
-    columns = [c for c in table.columns if c.name not in metadata_column_list]
+    columns = [c for c in table.columns
+               if c.name not in metadata_column_list
+               and not c.name.startswith(('_ref', '_mref'))]
+    reference_columns = [c for c in table.columns if c.name.startswith(('_ref', '_mref'))]
+
+    # Create the list of references
+    references = {}
+    for c in reference_columns:
+        # Column name is in the form of '_ref_attribute_name_catalog_collection'
+        column_name_array = c.name.split('_')
+        attribute_name = '_'.join(column_name_array[2:-2])
```

Comment:
I propose to add an example that explains this logic

Reply:
Docstring with explanation was added